weaviate/exceptions.py

```diff
@@ -224,15 +224,45 @@ def __init__(self, message: str):
 class WeaviateQueryError(WeaviateBaseError):
     """Is raised if a query (either gRPC or GraphQL) to Weaviate fails in any way."""
 
-    def __init__(self, message: str, protocol_type: str):
-        msg = f"""Query call with protocol {protocol_type} failed with message {message}."""
+    def __init__(self, message: str, protocol_type: str, add_message: bool = True):
+        if add_message:
+            msg = f"""Query call with protocol {protocol_type} failed with message {message}."""
+        else:
+            msg = message
         super().__init__(msg)
-        self.message = message
+        self.message = msg
 
 
 WeaviateQueryException = WeaviateQueryError
 
 
+class WeaviateQueryThirdPartyError(WeaviateQueryError):
+    """Is raised if a query (either gRPC or GraphQL) to Weaviate fails in any way."""
+
+    def __init__(
+        self,
+        full_error: str,
+        provider_name: str,
+        provider_error: str,
+        error_code: int,
+        request_id: str,
+        protocol_type: str,
+    ):
+        request_id_str = f"with request_id {request_id} " if request_id else ""
+
+        msg = f"""
+        Full error message: {full_error}
+
+        Query {request_id_str}failed due to an error with a third party service:
+        Provider: {provider_name}
+        Error code: {error_code}
+        Error from provider: {provider_error}
+
```

Comment:
Copyedit suggestion: Wdyt about:

```python
msg = f"""Query {request_id_str} failed:

Provider ({provider_name}) service error {error_code}.

Error details:

{'Provider:':>25} {provider_name}
{'Provider error code:':>25} {error_code}
{'Provider message:':>25} {provider_error}

Weaviate error message:

{full_error}"""
```

Mainly to really clarify what part of the message comes from the provider? (Also for right-aligning the structured part, which - not important either way)